File: ch1/is_unique/isunique.py
import sys

def isUnique(str):
    for i in range (0, len(str)):
        for j in range (i + 1, len(str)):
            if(str[i] == str[j]):
                return False
    return True

def main():
    if len(sys.argv) < 2:
        print("Use: $ python isunique.py word1 word2 word3 etc")
    else:
        for arg in sys.argv[1:]:
            print(isUnique(arg))

if __name__ == "__main__":
    main()

File: ch1/is_unique/isUnique.java
public class isUnique {
    public static boolean isUn(String word){
        for(int i = 0; i < word.length(); i++){
            for(int j = i + 1; j < word.length(); j++){
                if(word.charAt(i) == word.charAt(j)){
                    return false;
                }
            }
        }
        return true;
    }
    public static void main(String[] args){
        if(args.length < 1){
            System.out.println("Usage! java isUnique word1 word2 etc");
        } else {
            for(int i = 0; i < args.length; i++){
                System.out.println(isUn(args[i]));
            }
        }
    }
}
